packages/dev/core/src/Lights/LTC/ltcTextureTool.ts

```diff
@@ -0,0 +1,109 @@
+import { RawTexture } from "core/Materials/Textures/rawTexture";
+import type { BaseTexture } from "core/Materials/Textures/baseTexture";
+import { Texture } from "core/Materials/Textures/texture";
+import type { Scene } from "core/scene";
+import { Constants } from "core/Engines/constants";
+import { Tools } from "core/Misc/tools";
+import type { Tuple } from "core/types";
+
+/**
+ * Interface that can provide LTC textures used by area Lights. Users can override the default ones to provide their own LTC textures.
+ */
+export interface IAreaLightLTCProvider {
+    /**
+     * Linearly trasnformed cossine texture for BRDF.
+     */
+    ltc1Texture: BaseTexture;
+
+    /**
+     * Linearly trasnformed cossine texture for BRDF.
+     */
+    ltc2Texture: BaseTexture;
+}
+
+/**
+ * Default provider for LTC textures. This provider will load the LTC data from the Babylon CDN.
+ */
+export class DefaultAreaLightLTCProvider implements IAreaLightLTCProvider {
+    /**
+     * LTC Texture 1 loaded from the Babylon.js CDN.
+     */
+    public ltc1Texture: BaseTexture;
+
+    /**
+     * LTC Texture 2 loaded from the Babylon.js CDN.
+     */
+    public ltc2Texture: BaseTexture;
+
+    constructor(scene: Scene) {
+        this._buildSceneLTCTextures(scene);
+    }
+
+    // Loads LTC textures from CDN and assigns them to ltc1Texture and ltc2Texture textures.
+    private async _buildSceneLTCTextures(scene: Scene): Promise<void> {
+        const textureData = await this._decodeLTCTextureData();
+        this.ltc1Texture = this._createLTCTextureFromArray(textureData[0], scene);
+        this.ltc2Texture = this._createLTCTextureFromArray(textureData[1], scene);
+        for (const mesh of scene.meshes) {
+            if (mesh.lightSources.filter((a) => a.getClassName() === "RectAreaLight")) {
+                mesh._markSubMeshesAsLightDirty();
+            }
+        }
+
+        scene.onDisposeObservable.addOnce(() => {
+            this.ltc1Texture.dispose();
+            this.ltc2Texture.dispose();
+        });
+    }
+
+    // Loads raw binary data from CDN and assigns to the correct texture array data.
+    private async _decodeLTCTextureData(): Promise<Tuple<Uint16Array, 2>> {
+        const _ltc1 = new Uint16Array(64 * 64 * 4);
+        const _ltc2 = new Uint16Array(64 * 64 * 4);
+
+        const ltcPath = Tools.GetBabylonScriptURL("https://assets.babylonjs.com/areaLights/areaLightsLTC.bin", true);

```

Comment:
can the user change it? do we want to open that?

Reply:
Users can provide their own ```IAreaLightLTCProvider```. Their implementation can load the LTC data however they want. (They can even store as float array in JS if they want), the only thing the interface needs to provide are the two textures.

Reply:
I think we should still support something through the script base url if we can ? this is the centralized way for our users to prevent external call.